examples/experimental/dagster-airlift/dagster_airlift/in_airflow/dagster_run_utils.py

```diff
@@ -0,0 +1,41 @@
+from typing import Any, Mapping, NamedTuple
+
+TERMINAL_STATI = ["SUCCESS", "FAILURE", "CANCELED"]
+SYSTEM_TAG_PREFIX = "dagster/"
+MAX_RETRIES_TAG = f"{SYSTEM_TAG_PREFIX}max_retries"
+RETRY_NUMBER_TAG = f"{SYSTEM_TAG_PREFIX}retry_number"
+PARENT_RUN_ID_TAG = f"{SYSTEM_TAG_PREFIX}parent_run_id"
+SUCCESS_STATUS = "SUCCESS"
+
+
+class DagsterRunResult(NamedTuple):
+    status: str
+    tags: Mapping[str, Any]
+
+    @property
+    def run_retries_configured(self) -> bool:
+        return MAX_RETRIES_TAG in self.tags
+
+    @property
+    def has_remaining_retries(self) -> bool:
```

Comment:
this isn't totally accurate due to the retry_on_asset_or_op_failure setting: https://docs.dagster.io/deployment/run-retries#combining-op-and-run-retries - just because a run has max_retries set and has failed does not mean that it will always retry that many times - see https://github.com/dagster-io/dagster/blob/master/python_modules/dagster/dagster/_daemon/auto_run_reexecution/auto_run_reexecution.py#L24 for how this is currently calculated

Reply:
Ah; good call. I'll make sure to account for this.

Reply:
@gibsondan ; I added handling for the case mentioned here + a test.